WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldEditorView.swift

```diff
@@ -53,54 +32,64 @@ struct CustomFieldEditorView: View {
                         )
-                        .onChange(of: key) { _ in
-                            checkForModifications()
-                        }
                 }
 
                 // Value Input
                 VStack(alignment: .leading, spacing: Layout.subSectionsSpacing) {
                     HStack {
-                        Text("Value") // todo-13493: set String to be translatable
+                        Text(Localization.valueLabel)
                             .foregroundColor(Color(.text))
                             .subheadlineStyle()
 
                         Spacer()
 
                         if !isReadOnlyValue {
-                            Button(action: {
-                                showRichTextEditor = true
-                            }) {
-                                Text("Edit in Rich Text Editor") // todo-13493: set String to be translatable
-                                    .font(.footnote)
+                            Picker(Localization.editorPickerLabel, selection: $showRichTextEditor) {
+                                Text(Localization.editorPickerText).tag(false)
+                                Text(Localization.editorPickerHTML).tag(true)
                             }
-                            .buttonStyle(.plain)
-                            .foregroundColor(Color(uiColor: .accent))
+                            .pickerStyle(.segmented)
+                            .fixedSize()
                         }
                     }
 
-                    TextEditor(text: isReadOnlyValue ? .constant(value) : $value)
-                        .bodyStyle()
+                    if showRichTextEditor {
+                        AztecEditorView(initialValue: value,
+                                        onContentChanged: { content in
+                            DispatchQueue.main.async {
+                                value = content
+                            }
+                        })
                         .frame(minHeight: Layout.minimumEditorSize)
+                        .clipped()
                         .padding(insets: Layout.inputInsets)
                         .background(Color(.listForeground(modal: false)))
                         .overlay(
                             RoundedRectangle(cornerRadius: Layout.cornerRadius).stroke(Color(.separator))
                         )
-                        .onChange(of: value) { _ in
-                            checkForModifications()
-                        }
+                    } else {
+                        TextEditor(text: isReadOnlyValue ? .constant(value) : $value)
+                            .foregroundColor(Color(.text))
+                            .subheadlineStyle()
+                            .frame(minHeight: Layout.minimumEditorSize)
+                            .padding(insets: Layout.inputInsets)
+                            .background(Color(.listForeground(modal: false)))
```

Comment:
Since there are rounded corners on the overlay, the background view should have rounded corners as well. If you look closely, the background view is currently showing underneath which doesn't look nice.

<img width="134" alt="image" src="https://github.com/user-attachments/assets/bf99f566-b73a-45d5-8451-fd7432f3a9ce">

Same comment for the other fields.

Reply:
Wow, great eye! So these fields are pretty much copied from `SupportForm` (Settings > Help > Contact Support) as they're the only View I could find that have more than one text input fields in one screen in the app. 

I see that it also has the same problem:
<img width="438" alt="Screenshot 2024-09-24 at 18 55 18" src="https://github.com/user-attachments/assets/af22b622-7baa-424b-a501-69586f03dbfd">

I'll handle this in a separate issue to fix both views at once.

Reply:
#14028